src/main/java/ubc/pavlab/rdp/util/PurlResolver.java

```diff
@@ -0,0 +1,76 @@
+package ubc.pavlab.rdp.util;
+
+import lombok.extern.apachecommons.CommonsLog;
+import org.springframework.beans.BeansException;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.ApplicationContextAware;
+import org.springframework.context.ResourceLoaderAware;
+import org.springframework.context.support.GenericApplicationContext;
+import org.springframework.core.io.*;
+import org.springframework.stereotype.Component;
+import sun.net.www.content.text.Generic;
+import ubc.pavlab.rdp.model.Gene;
+
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.URL;
+
+/**
+ * Resolve resources from <a href="http://purl.obolibrary.org">purl.obolibrary.org</a>.
+ *
+ * @author poirigui
+ */
+@Component
+@CommonsLog
+public class PurlResolver implements ProtocolResolver, ResourceLoaderAware, ApplicationContextAware {
+
+    /**
+     * Check if the provided URL refers to a PURL resource.
+     */
+    public static boolean isPurl( URL url ) {
+        return url.getProtocol().equals( "http" ) && url.getHost().equals( "purl.obolibrary.org" );
+    }
+
+    @Override
+    public Resource resolve( String location, ResourceLoader resourceLoader ) {
+        HttpURLConnection con = null;
+        try {
+            URL url = new URL( location );
+            if ( isPurl( url ) ) {
+                con = (HttpURLConnection) url.openConnection();
+                con.setInstanceFollowRedirects( false );
+                if ( con.getHeaderField( "Location" ) != null ) {
+                    return new UrlResource( con.getHeaderField( "Location" ) );
+                }
+            }
+        } catch ( MalformedURLException e ) {
+            return null;
+        } catch ( IOException e ) {
+            log.error( "Failed to resolve %s.", e );
+            if ( con != null ) {
```

Comment:
IntelliJ is suggesting that this check is unnecessary, but it is. If `url.openConnection()` fails, it can indeed be null.